cpp/include/cudf/utilities/traits.hpp

```diff
@@ -544,6 +544,72 @@ constexpr inline bool is_fixed_width(data_type type)
   return cudf::type_dispatcher(type, is_fixed_width_impl{});
 }
 
+/**
+ * @brief Indicates whether the type `T` is a lists type.
+ *
+ * @tparam T  The type to verify
+ * @return true `T` is a lists-type
+ * @return false  `T` is not lists-type
+ */
+template <typename T>
+constexpr inline bool is_lists()
+{
+  return std::is_same<list_view, T>::value;
+}
+
+struct is_lists_impl {
+  template <typename T>
+  constexpr bool operator()()
+  {
+    return is_lists<T>();
+  }
+};
+
+/**
+ * @brief Indicates whether `type` is a lists `data_type`.
+ *
+ * @param type The `data_type` to verify
+ * @return true `type` is a lists type
+ * @return false `type` is not a lists type
+ */
+constexpr inline bool is_lists(data_type type)
+{
+  return cudf::type_dispatcher(type, is_lists_impl{});
+}
+
+/**
+ * @brief Indicates whether the type `T` is a structs type.
+ *
+ * @tparam T  The type to verify
+ * @return true `T` is a structs-type
+ * @return false  `T` is not structs-type
+ */
+template <typename T>
+constexpr inline bool is_structs()
```

Comment:
I'm not sure about the plural "s" here and with `is_lists`. All the other traits are singular like `is_fixed_width` or `is_chrono`.

Reply:
Right, it probably make more sense align with other namings in `traits.hpp`.

I got the plural form idea from `lists_column_view` and `structs_column_view` and was under the impression they were named that way since.

Reply:
Why not`is_list_view`

Reply:
The below reveals the fact that we use `list_view` as an alias for `LIST` type during type-dispatching, which is too detailed IMO.
```c++
template <typename T, CUDF_ENABLE_IF(is_list_view<T>())>
```
while
```c++
template <typename T, CUDF_ENABLE_IF(is_list<T>())>
```
seems just clear enough.

Reply:
But `LIST` is not a C++ type, and `is_X()` is a type trait.  After all, `is_list<T>()` is just syntactic sugar for `std::is_same<T, list_view>`, so it's barely needed at all.

Reply:
Previously the code was doing `rhs.type().id()==type_id::LIST` and `rhs.type().id()==type_id::STRUCT` so the thought was to make `is_list(data_type)` and `is_struct(data_type)` functions. Since the code is now using the type-dispatcher, these are no longer required since we can just do `std::is_same`.

Reply:
We might be able to get rid of `is_same` in all if we use explicit specialization:
```
template <>
bool columns_equal_fn::operator()<list_view>(column_view const& lhs, column_view const& rhs)
```